src/geometry/manifold/ManifoldGeometry.cc

```diff
@@ -76,17 +78,15 @@ std::string ManifoldGeometry::dump() const {
 }
 
 std::shared_ptr<const PolySet> ManifoldGeometry::toPolySet() const {
-  auto ps = std::make_shared<PolySet>(3);
   manifold::Mesh mesh = getManifold().GetMesh();
-  ps->reserve(mesh.triVerts.size());
-  Polygon poly(3);
-  for (const auto &tv : mesh.triVerts) {
-    for (const int j : {0, 1, 2}) {
-      poly[j] = vector_convert<Vector3d>(mesh.vertPos[tv[j]]);
-    }
-    ps->append_poly(poly);
-  }
-  return ps;
+  std::vector<int> indmap;
+  indmap.reserve(mesh.vertPos.size());
+  PolySetBuilder builder(mesh.vertPos.size(),mesh.triVerts.size());
```

Comment:
You should not use the builder with manifold. It purposely duplicate the vertices to maintain manifoldness, so deduplicating them will just cause you more trouble. Also, the mesh data structure is very much like the indexed polyset, so there is no point hashing everything again...